src/plugins/auto/remote_context.hpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief a header that defines wrappers for internal GPU plugin-specific
+ * OpenCL context and OpenCL shared memory blobs
+ *
+ * @file gpu_context_api_ocl.hpp
+ */
+#pragma once
+
+#include "ie_remote_context.hpp"
+#include "ie_core.hpp"
+#include <memory>
+#include <string>
+
+namespace MultiDevicePlugin {
+
+/**
+ * @brief This class represents an abstraction for Multi plugin remote context
+ * which is wrapper of underline hardware remote contexts.
+ * The plugin object derived from this class can be obtained using CreateContext() Core call.
+ */
+class MultiRemoteContext : public RemoteContext {
+public:
+    /**
+     * @brief A smart pointer to the MultiRemoteContext object
+     */
+    using Ptr = std::shared_ptr<MultiRemoteContext>;
+
+    MultiRemoteContext() = default;
+    MultiRemoteContext(const std::string pluginName) : m_plugin_name(pluginName) {}
+
+    RemoteBlob::Ptr CreateBlob(const TensorDesc& tensorDesc, const ParamMap& params = {}) override {
+        std::lock_guard<std::mutex> locker(m_mutex);
+        if (m_contexts.size() == 0) {
+            IE_THROW() << "no valid context available";
+        }
+        return m_contexts[0]->CreateBlob(tensorDesc, params);
+    }
+
+    ParamMap getParams() const override {
```

Comment:
multi-context params should probably contain single element like:
`{"HW_CONTEXTS", std::vector<RemoteContext>{}}`